src/server/webkit/wkPage.ts

```diff
@@ -1160,3 +1195,62 @@ function isLoadedSecurely(url: string, timing: network.ResourceTiming) {
     return true;
   } catch (_) {}
 }
+
+
+class WKNavigation {
+  private readonly _page: WKPage;
+  // loaderId is a navigation id which is unique within browser and persists during cross-process navigation.
+  private readonly _loaderId: string;
+  readonly _originalRequestId: Protocol.Network.RequestId;
+  private readonly _frameId: string;
+  private _responseReceived: boolean = false;
+  private _didCheckFirstRequestInProvisionalPage: boolean = false;
+  private _provisionalPageRequestId: string | undefined;
+
+  constructor(page: WKPage, event: Protocol.Network.requestWillBeSentPayload) {
+    this._page = page;
+    this._loaderId = event.loaderId;
+    this._originalRequestId = event.requestId;
+    this._frameId = event.frameId;
+  }
+
+  shouldIgnoreFrameStoppedLoading(frameId: string): boolean {
+    if (frameId !== this._frameId)
+      return false;
+    // Navigation in the original frame is canceled and actually continues in the
```

Comment:
I am afraid this might not be always the case. For example, we navigate to cross-origin 204 page that creates a provisional page but never commits?

Reply:
Coincidentally that works because we generate loadingFinished event in WebKit ourselves when 204 response comes: https://github.com/microsoft/playwright/blob/45b365d95833e0ab61bd49e0b98293542fbd12c0/src/server/webkit/wkPage.ts#L1023-L1030

With network failures we will receive `Playwright.provisionalLoadFailed` and that will abort the navigation, so we are good in that case too. Added a test.